src/tools/pub/glossary.md

```diff
@@ -12,11 +12,10 @@ the [pub tool][].
 
 ## Application package
 
-A package that is not intended to be used as a library. Application packages may
-have [dependencies][] on other packages, but are never depended on
-themselves. They are usually meant to be run directly, either on the command
-line or in a browser. The opposite of an application package is a
-[library package][].
+A package that is not intended to be shared, unlike regular [packages][].
```

Comment:
I think I'd start with what this is, rather than what this isn't. Maybe something like:

---

A package that contains a program or app, with a main entrypoint in `bin`. They are meant to be run directly, either on the command line or in a browser. 
 
Application packages may have [dependencies][] on other packages, but are never
depended on themselves.